src/cmake/modules/CUnit/src/main.c.in

```diff
@@ -9,23 +9,224 @@
+        if (*pat == '?') {
+            /* any character will do */
+            if (*str++ == 0) {
+                return 0;
+            }
+            pat++;
+        } else if (*pat == '*') {
+            /* collapse a sequence of wildcards, requiring as many
+               characters in str as there are ?s in the sequence */
+            while (*pat == '*' || *pat == '?') {
+                if (*pat == '?' && *str++ == 0) {
+                    return 0;
+                }
+                pat++;
+            }
+            /* try matching on all positions where str matches pat */
+            while (*str) {
+                if (*str == *pat && patmatch(pat+1, str+1)) {
+                    return 1;
+                }
+                str++;
+            }
+            return *pat == 0;
+        } else {
+            /* only an exact match */
+            if (*str++ != *pat++) {
+                return 0;
+            }
+        }
+    }
+
+    return *str == 0;
+}
+
+static void usage(const char *name)
+{
+    fprintf(stderr, "Usage: %s OPTIONS\n", name);
+    fprintf(stderr, "Try '%s -h' for more information\n", name);
+}
+
+static void help(const char *name)
+{
+    printf("Usage: %s [OPTIONS]\n", name);
+    printf("\n");
+    printf("Possible options:\n");
+    printf("  -a           run in automated mode\n");
+    printf("  -f           fail fast\n");
+    printf("  -h           display this help and exit\n");
+    printf("  -j           junit format results \n");
+    printf("  -r FILENAME  results file for automated run\n");
+    printf("  -s PATTERN   run only tests in suites matching pattern\n");
+    printf("  -t PATTERN   run only test matching pattern\n");
+}
+
+static CU_ErrorCode parse_options(int argc, char *argv[])
+{
+    CU_ErrorCode err = CUE_SUCCESS;
+
+
+    for (int i = 1; err == CUE_SUCCESS && i < argc; i++) {
+        switch ((argv[i][0] == '-') ? argv[i][1] : 0) {
+            case 'a':
+                opts.automated = true;
+                break;
+            case 'f':
+                opts.error_action = CUEA_FAIL;
+                break;
+            case 'h':
+                opts.print_help = true;
+                break;
+            case 'j':
+                opts.junit = true;
+                break;
+            case 'r':
+                if((i+1) < argc){
+                    opts.results = argv[++i];
+                    break;
+                }
+                /* FALLS THROUGH */
+            case 's':
+                if ((i+1) < argc) {
+                    opts.suite = argv[++i];
+                    break;
+                }
+                /* FALLS THROUGH */
+            case 't':
+                if ((i+1) < argc) {
+                    opts.test = argv[++i];
+                    break;
+                }
+                /* FALLS THROUGH */
+            default:
+                err = (CU_ErrorCode)256;
+                CU_set_error(err); /* Will print as "Undefined Errpr" */
```

Comment:
No need for ``CU_set_error``: you're exiting the process immediately anyway, so no need to worry about what ``CU_ErrorCode`` to use.